charmstore/lib.py

```diff
@@ -115,10 +109,18 @@ def __init__(self, id=None, api=None, timeout=None):
             )
 
     def revisions(self):
-        data = self.raw.get('revision-info', {}).get('Revisions', [])
+        if self.revisions is None:
+            self.revisions = self.theblues._meta(self.id, ['revision-info']) \
+                or {}
+        data = self.revisions.get('Revisions', [])
         return [self.__class__(e) for e in data]
 
     def file(self, path):
+        if self._files_fetched is False:
+            self.files = [
+                f.get('Name') for f in
+                    self.theblues._meta(self.id, ['manifest']) ]
```

Comment:
I take it the response to this call doesn't include the contents?

Reply:
the `_meta` call contains the actual HTTP request to the charmstore. What this line is doing is asking the charmstore for the "manifest" field of the given `Entity`

Reply:
Yeah, sorry I was unclear.  I just wanted to confirm that the response from the `_meta` call didn't also contain the contents that we were throwing away here only to re-request subsequently.  Pretty sure that wouldn't be the case because that would be too much data, but you never know.